src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.OSX.cs

```diff
@@ -304,7 +256,9 @@ private void WatchForFileSystemEventsThreadStart()
                         long pairedId = FindRenameChangePairedChange(i, eventPaths, eventFlags, eventIds);
                         if (pairedId == long.MinValue)
                         {
-                            OnError(new ErrorEventArgs(new ArgumentOutOfRangeException("pairedId")));
+                            // Getting here means we have a rename without a pair, meaning it should be a create for the 
+                            // move from unwatched folder to watcher folder scenario
+                            NotifyFileSystemEventArgs(WatcherChangeTypes.Created, relativePath);
```

Comment:
Could it also be the opposite, that we get a rename for the "from" but not for the "to" because the file moved out of the direction being watched rather than into the directory being watched?